src/language_model/pipeline.py

```diff
@@ -0,0 +1,76 @@
+import os
+import pathlib
+import sys
+from abc import ABC
+from logging import INFO, FileHandler, Formatter, Logger, StreamHandler, getLogger
+from typing import Sequence
+
+sys.path.append(str(pathlib.Path().absolute()))
+
+LOGGING_FORMAT: str = "%(asctime)s : %(levelname)s : %(module)s : %(message)s"
+DEFAULT_LOG_DIR: str = "logs"
+DEFAULT_LOG_FILE: str = "log.txt"
+DEFAULT_CONFIGURATION_DIR: str = "configs"
+
+
+class ITask(object):
+    def execute(self, environment_path: str) -> None:
+        raise NotImplementedError()
+
+
+class Sandbox(object):
+    def get_sandbox_folder_path(self) -> str:
+        raise NotImplementedError()
+
+
+class AbstractSandbox(Sandbox):
+    def __init__(self, sandbox_folder_path: str) -> None:
+        self.sandbox_folder_path = sandbox_folder_path
+
+    def get_sandbox_folder_path(self) -> str:
+        return self.sandbox_folder_path
+
+
+class SandboxTask(ITask, AbstractSandbox, ABC):
```

Comment:
Why `Sandbox`? This looks like running tasks that extend this class will not actually save/do/process anything, but rather test something.

Reply:
Do you associate the `sandbox` term with testing? If this association is strong we would want to find another name for a place where usually experiments are run **in isolation** - I followed this semantics when named it that way.

Reply:
I guess, the fact that sandboxes are used for testing, does not make them test utilities only. The concept must be more generic, afaic

Reply:
I agree that this sandbox only set the usage of filesystem (pointing to a directory), but not the whole system (devices, ram, etc.)

Reply:
My association is the following: Sandbox - some isolated new environment used mostly for harmlessly testing things that might otherwise influence or irreversibly change some parts of the existing environment which we want to stay the same. This Sandbox can easily be destroyed without leaving a trace of its existence apart from maybe some logs.